django_webapps/scripts/db_population/parsers/common_parsers.py

```diff
@@ -0,0 +1,58 @@
+import contextlib
+import datetime
+import re
+
+import pytz
+from geopy.geocoders import Nominatim
+from timezonefinder import TimezoneFinder
+
+
+def _city_to_timezone(city_name: str) -> str | None:
+    geolocator = Nominatim(user_agent="city-to-timezone")
+    coords = geolocator.geocode(city_name)
+    tf = TimezoneFinder()
+    if coords:
+        return tf.timezone_at(lng=coords.longitude, lat=coords.latitude)
+
+    return None
+
+
+def parse_timezone(tz: str) -> datetime.tzinfo | None:
+    tz_str = str(tz)
+    tz_name = None
+    offset = None
+    regex_right = r"((UTC|GMT)([\+|\-]\d*))"
+    regex_left = r"(([\+|\-]\d*)(UTC|GMT))"
+    tz_str = str(tz).replace(" ", "").upper()
+
+    match_result_right = re.findall(regex_right, tz_str)
+    match_result_left = re.findall(regex_left, tz_str)
+    if len(match_result_right) > 0:
+        offset, tz_name = match_result_right[0][2], match_result_right[0][1]
+    elif len(match_result_left) > 0:
+        offset, tz_name = match_result_left[0][1], match_result_left[0][2]
```

Comment:
findall searches for occurrences in the whole string, I also like it since, for example, `re.search` returns you an object but here you have a list of matches, we can take out the first group and assign it to a variable